src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -16,68 +19,65 @@ namespace System.Data.SqlTypes
     [Serializable]
     [XmlSchemaProvider("GetXsdType")]
     [System.Runtime.CompilerServices.TypeForwardedFrom("System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    public struct SqlGuid : INullable, IComparable, IXmlSerializable, IEquatable<SqlGuid>
+    public struct SqlGuid : INullable, IComparable, ISerializable, IXmlSerializable, IEquatable<SqlGuid>
     {
-        private const int SizeOfGuid = 16;
+        private const int SizeOfGuid = 16; // sizeof(Guid)
 
         // NOTE: If any instance fields change, update SqlTypeWorkarounds type in System.Data.SqlClient.
-        private byte[]? m_value; // the SqlGuid is null if m_value is null
+        private Guid? _value; // the SqlGuid is null if _value is null
 
         // constructor
         // construct a SqlGuid.Null
         private SqlGuid(bool fNull)
```

Comment:
The singleton readonly field can be initialized to `default`, which would allow this ctor to go away entirely.